--- /tmp/gemm_a8w8.cu.orig	2026-09-15 17:38:26.386812673 +0800
+++ /home/qiba/ai/envs/wu1w-int8-028/lib/python3.12/site-packages/aiter_meta/csrc/ck_gemm_a8w8/gemm_a8w8.cu	2026-09-15 17:38:26.388812691 +0800
@@ -30,27 +30,27 @@
   {
     return a8w8_rowwise_256x256x128x64_32x32_4x2_4x64x1_4x64x1_1x32x1x8_8x8x1_1x1_interwave_v1<ABDataType, DDataType, EDataType>;
   }
-  else if (M < 64 && N < 2048 && K < 2048)
+  else if (M <= 64 && N < 2048 && K < 2048)
   {
     // Kernel that generally works well on small shapes.
     return a8w8_rowwise_64x16x16x128_16x16_1x1_8x8x1_8x8x1_1x16x1x4_4x4x1_1x1_interwave_v2<ABDataType, DDataType, EDataType>;
   }
-  else if (M < 64 && K < 2048)
+  else if (M <= 64 && K < 2048)
   {
     // Kernel that works well for small batch size and small K.
     return a8w8_rowwise_128x16x32x128_16x16_1x1_8x16x1_8x16x1_1x16x1x8_4x4x1_1x1_intrawave_v2<ABDataType, DDataType, EDataType>;
   }
-  else if (M < 64 && N < 2048)
+  else if (M <= 64 && N < 2048)
   {
     // Kernel that works well for small batch size and small N.
     return a8w8_rowwise_128x32x16x128_16x16_1x1_8x16x1_8x16x1_1x16x1x8_2x2x1_1x1_interwave_v2<ABDataType, DDataType, EDataType>;
   }
-  else if (M < 64 && N > 2048 && K > 2048)
+  else if (M <= 64 && N > 2048 && K > 2048)
   {
     // Kernel that works well for small M but larger N and K.
     return a8w8_rowwise_64x16x16x256_16x16_1x1_16x4x1_16x4x1_1x16x1x4_4x4x1_1x1_intrawave_v1<ABDataType, DDataType, EDataType>;
   }
-  else if (M < 64)
+  else if (M <= 64)
   {
     // Fallback to generic small batch kernel if we cant find a good match.
     return a8w8_rowwise_64x16x16x128_16x16_1x1_8x8x1_8x8x1_1x16x1x4_4x4x1_1x1_interwave_v2<ABDataType, DDataType, EDataType>;
